python/cuml/fil/README.md

```diff
@@ -0,0 +1,77 @@
+# FIL - RAPIDS Forest Inference Library
+
+The Forest Inference Library provides a lightweight, flexible API to
+infer (predict) results from a tree-based model ensemble on GPU. The
+tree ensemble can be either gradient-boosted decision tree (GBDT) or
+random forest (RF) models trained in XGBoost, cuML, scikit-learn, or
+LightGBM.
+
+# Code sample
+
+Starting with an XGBoost classification model saved in the file
+"xgb.mod," we want to use that model to infer on a large dataset of
+test samples.
+
+```python
+from cuml import ForestInference
+
+fm = ForestInference.load(filename=model_path,
+                          output_class=True,
+                          threshold=0.50,
+                          model_type='xgboost')
+
+X = ... load test samples as a numpy or cupy array ...
+
+y_out = fm.predict(X)
+
+```
+
+See [the sample notebook](https://github.com/rapidsai/cuml/blob/branch-0.16/notebooks/forest_inference_demo.ipynb) for much more detail and runnable samples.
+
+Additionally, FIL can be called directly from C or C++ code. See [the API docs here](https://docs.rapids.ai/api/libcuml/nightly/namespaceML_1_1fil.html)
```

Comment:
I wonder if we should add somewhere that we're planning to move the functions other than `from_treelite()`, `free()` and `predict()` to an internal header, and thus it's best not to rely on them.

Reply:
Can that be added to the function documentation itself?

Reply:
I would add function-level notes to the functions themselves. I think the intended audience for this doc is beginners.